proposed/http-client/http-client.md

```diff
@@ -7,43 +7,63 @@ The key words "MUST", "MUST NOT", "REQUIRED", "SHALL", "SHALL NOT", "SHOULD",
 "SHOULD NOT", "RECOMMENDED", "MAY", and "OPTIONAL" in this document are to be
 interpreted as described in [RFC 2119](http://tools.ietf.org/html/rfc2119).
 
-## Specification
+## Goal
 
-### Client
+The goal of this PSR is to allow developers to create libraries decoupled from HTTP client
+implementations. This will make libraries more reusable as it reduces the number of
+dependencies and lowers the likelihood of version conflicts.
 
-An HTTP client has the responsibility to send a PSR-7 request and return a PSR-7
-response. Under the hood, the HTTP client MAY modify the request received from
-the user and/or the response received from the server. In this case, the request
-and the response MUST be consistent between the body and headers. For example, a
-server may return a gzip-encoded body and the client may decide to decode the
-body. If the client decodes the body, the client MUST also remove the
-`Content-Encoding` header and adjust the `Content-Length` header.
+A second goal is that HTTP clients can be replaced as per the
+[Liskov substitutions principle][Liskov]. This means that all clients MUST behave in the
+same way when sending a request.
 
-### Exceptions
+## Definitions
 
-All exceptions thrown by the client MUST implement `Psr\Http\Client\ClientExceptionInterface`.
+* **Client** - A Client is a library that implements this specification for the purposes of
+sending PSR-7-compatible HTTP Request messages and returning a PSR-7-compatible HTTP Response message to a Caller.
+* **Caller** - A Caller is any code that makes use of a Client.
```

Comment:
I think in PSR-6 we called this the "calling library", so maybe we should use that term here as well?

Reply:
Dont we use "consuming library" or something like that?

Reply:
but its not restricted to libraries, it could also be an application that is using a PSR HTTP client. seems fine to me as you put it now.

Reply:
This is why a definitions section is useful, to define precisely what "library" means. ;-)

Here's what we used in PSR-6, for reference:

```
Calling Library - The library or code that actually needs the cache
services. This library will utilize caching services that implement this
standard’s interfaces, but will otherwise have no knowledge of the
implementation of those caching services.
```

Reply:
cool if you can add this, adapted for sending requests instead of caching!

Reply:
Pushed a new commit that changes `Caller` to `Calling Library` and tweaks the definition a bit.